vm/src/hint_processor/builtin_hint_processor/bootloader/execute_task_hints.rs

```diff
@@ -172,4 +231,71 @@ mod tests {
             expected_program_size
         );
     }
+
+    #[rstest]
+    fn test_append_fact_topologies(fibonacci: Program) {
+        let task = Task {
+            program: fibonacci.clone(),
+        };
+
+        let mut vm = vm!();
+
+        // Allocate space for the pre-execution and return builtin structs (2 x 8 felts).
+        // The pre-execution struct starts at (1, 0) and the return struct at (1, 8).
+        // We only set the output values to 0 and 10, respectively, to get an output size of 10.
+        vm.segments = segments![((1, 0), 0), ((1, 8), 10),];
+        vm.run_context.fp = 16;
+        add_segments!(vm, 1);
+
+        let tree_structure = vec![1, 2, 3, 4];
+        let program_output_data = OutputBuiltinAdditionalData {
+            base: 0,
+            pages: HashMap::from([
+                (1, PublicMemoryPage { start: 0, size: 7 }),
+                (2, PublicMemoryPage { start: 7, size: 3 }),
+            ]),
+            attributes: HashMap::from([("gps_fact_topology".to_string(), tree_structure.clone())]),
+        };
+        let mut output_builtin = OutputBuiltinRunner::new(true);
+        output_builtin.set_state(program_output_data.clone());
+        output_builtin.initialize_segments(&mut vm.segments);
+        vm.builtin_runners
+            .push(BuiltinRunner::Output(output_builtin));
+
+        let ids_data = non_continuous_ids_data![
+            ("pre_execution_builtin_ptrs", -16),
+            ("return_builtin_ptrs", -8),
```

Comment:
`vars::` ?

Reply:
Actually we could introduce an `ids` module to separate the two?